File: Python/leetcode/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        d = {}
        for i in range(len(nums)):
            d[target - nums[i]] = i
        for i in range(len(nums)):
            if nums[i] in d.keys() and i != d[nums[i]]:
                if i > d[nums[i]]:
                    ans = [d[nums[i]], i]
                else:
                    ans = [i, d[nums[i]]]
                break
        return ans

# @lc code=end

File: Java/1.两数之和.java
/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(nums[i])) {
                int left = map.get(nums[i]);
                return new int[] { left, i };
            } else {
                map.put(target - nums[i], i);
            }
        }
        return null;
    }
}
// @lc code=end
